api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -947,6 +981,20 @@ type ProtoPort struct {
 	Net string `json:"net,omitempty"`
 }
 
+// +kubebuilder:validation:Pattern=`^(([0-9]*(\.[0-9]*)?(ms|s|h|m|us)+)+|auto)$`
```

Comment:
```suggestion
// +kubebuilder:validation:Pattern=`^(([0-9]*(\.[0-9]*)?(ms|s|h|m|us)+)+|Auto)$`
```

I think this should be "Auto" based on k8s API conventions for constant values: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants